github/commitstatus.go

```diff
@@ -111,18 +113,23 @@ func (s CommitStatus) Add(sha, state, targetURL, description string) error {
 	var response httpResponse
 	timeToSleep := 0 * time.Second // 0 seconds
 
+retryLoop:
 	for attempt := 1; attempt <= s.target.MaxRetries; attempt++ {
 		time.Sleep(timeToSleep)
 		s.log.Info("GitHub HTTP request", "attempt", attempt, "max", s.target.MaxRetries)
 		response, err = httpRequestDo(req)
 		if err != nil {
 			return err
 		}
-		timeToSleep = checkForRetry(response, s.target.WaitTime, s.target.MaxSleepTime)
-		if timeToSleep == 0 {
-			break
+		timeToSleep = checkForRetry(response, s.target.WaitTime, s.target.MaxSleepTime, s.target.Jitter)
+		switch {
+		case timeToSleep == 0:
+			break retryLoop
```

Comment:
@marco-m-pix4d I didn't find a better way to break a loop. Is this in go language spirit?

Reply:
if you use an `if` as opposed to a `switch`, you can use the `break` as before without the label at line 116, although the code as-is is correct.

But, I have the feeling we can simplify all this. Let me think a moment.